Aiven TypeScript template.

dewan-ahmed opened the pull request:
This PR adds a TypeScript template for Aiven. Currently, there is one Aiven template available for Go.

A related PR is also open which adds a Python template for Aiven: https://github.com/pulumi/templates/pull/479

cnunciato commented on aiven-typescript/index.ts:
```diff
@@ -0,0 +1,25 @@
+            kafkaConnect: "true",
+        },
+    },
+});
```
Would it make sense to export some results here -- maybe service host and port?

```suggestion
});

// Export the service host and port.
export const serviceHost = kafka.serviceHost;
export const servicePort = kafka.servicePort;
```

cnunciato reviewed (commented):
Thanks for the contribution! 🎉 Here's some initial feedback for ya.

dewan-ahmed commented on aiven-typescript/index.ts:
```diff
@@ -0,0 +1,25 @@
+import * as aiven from "@pulumi/aiven";
+
+const kafka1 = new aiven.Kafka("kafka1", {
+    project: "<YOUR_AIVEN_PROJECT_NAME>",
+    cloudName: "google-europe-west1",
+    plan: "business-4",
+    serviceName: "kafka-gcp-eu",
```
Makes sense @cnunciato 

Only comment I have is that it's difficult to use one default project/cloud/plan/serviceName that'll apply to most. I propose to leave out the default values and have folks set them explicitly.

cnunciato replied:
Absolutely, yep -- makes sense to leave out the `default` for that one.

dewan-ahmed commented:
Thanks for the review @cnunciato 🙏🏽 

I addressed the comments you made. Few comments/questions:

1. I removed the optional settings for this example Kafka service and only kept the four mandatory variables.
2. Where does the exported variables end up? I proposed adding service.URI as an exported variable but URI includes service credentials. Is this Ok from a security perspective?
3. The [comment](https://github.com/pulumi/templates/pull/480#discussion_r1068460261) on not having default values for these configs. What's your thought?

Looking forward to the next review.

cnunciato commented:
>  2. Where does the exported variables end up? I proposed adding service.URI as an exported variable but URI includes service credentials. Is this Ok from a security perspective?

I noticed this as well during testing -- the exported `serviceURI` seems to be tracked as a `[secret]`). In general this is safe, yes ([Pulumi secrets](http://pulumi-docs-origin-pr-8133-7f623a1c.s3-website.us-west-2.amazonaws.com/docs/intro/concepts/secrets/) are encrypted) -- however, is this value actually a secret? When I examined it, it appeared to be just host and a port, both of which appear to be plain strings; I don't see any actual credentials in there. Maybe I'm missing something? Or maybe the URL does contain credentials in certain circumstances that don't manifest here? 
 
>  3. The [comment](https://github.com/pulumi/templates/pull/480#discussion_r1068460261) on not having default values for these configs. What's your thought?

Yep, agree `project` probably doesn't make sense. But the others seem like they at least _could_ have defaults (at least given my admittedly minimal knowledge of Aiven), and in general we try to make our templates work as much "out of the box" we can. Let me know what you think, though -- definitely flexible and open to conforming to what Aiven users will likely expect here.

dewan-ahmed commented:
I spun up a Aiven for Redis instance and `rediss://default:AVNS_zqkhIoq6vwI2XJpAv6f@dewan-pulumi-redis-dev-advocates.aivencloud.com:13040` is the serviceURI I see (this service is already deleted if the credentials in plaintext is giving security folks anxiety attack 😁 ). here `default` is the username and the password is followed by the `:`. Is this what you see when observing the `serviceURI` field of the service created using Pulumi?

I agree with you on setting default values for cloud, plan, serviceName for this example/template since that'll enable our users to create the service faster. I'll make another commit.

cnunciato commented:
>Is this what you see when observing the serviceURI field of the service created using Pulumi?

@dewan-ahmed I saw that the template emits a `serviceURI` that's marked `[secret]`, but when I passed `--show-secrets` to `pulumi stack output` to see what the secret value was, the Kafka service didn't reveal anything. Good to know Redis does, though. 

In this case, I'd probably recommend (as in [my suggestion](https://github.com/pulumi/templates/pull/480#discussion_r1067578728)) emitting the host and port, just so the template user has something gets back something useful when the deployment completes. What do you think?

dewan-ahmed commented:
Agreed @cnunciato 

https://github.com/pulumi/templates/pull/480/commits/ee56a19508027a52dc5677732c041e6ef210e6a5 adds those two fields as you suggested 👍🏽

cnunciato reviewed (approved):
This looks great. 🙌  Thanks @dewan-ahmed! 

I'll take a look at your Python PR next.

cnunciato commented:
/run-templates
